cmd/flags.go

```diff
@@ -1,40 +1,48 @@
 package cmd
 
-import "flag"
+import (
+	"github.com/spf13/pflag"
+)
 
-var configFlags = []*flag.Flag{
+var configFlags = []*pflag.Flag{
 	loglevelFlag,
 	nodeConfigFlag,
 	trustedHashFlag,
 	trustedPeerFlag,
 	coreRemoteFlag,
+	p2pKeyFlag,
 }
 
 var (
-	loglevelFlag = &flag.Flag{
+	loglevelFlag = &pflag.Flag{
 		Name:     "log.level",
 		Usage:    "DEBUG, INFO, WARN, ERROR, DPANIC, PANIC, FATAL, and\n// their lower-case forms",
 		DefValue: "info",
 	}
-	nodeConfigFlag = &flag.Flag{
+	nodeConfigFlag = &pflag.Flag{
 		Name:     "node.config",
 		Usage:    "Path to a customized Config",
 		DefValue: "",
 	}
-	trustedHashFlag = &flag.Flag{
+	trustedHashFlag = &pflag.Flag{
 		Name:     "headers.trusted-hash",
 		Usage:    "Hex encoded block hash. Starting point for header synchronization",
 		DefValue: "",
 	}
-	trustedPeerFlag = &flag.Flag{
+	trustedPeerFlag = &pflag.Flag{
 		Name:     "headers.trusted-peer",
 		Usage:    "Multiaddr of a reliable peer to fetch headers from",
 		DefValue: "",
 	}
-	coreRemoteFlag = &flag.Flag{
+	coreRemoteFlag = &pflag.Flag{
 		Name: "core.remote",
 		Usage: "Indicates node to connect to the given remote core node. " +
 			"Example: <protocol>://<ip>:<port>, tcp://127.0.0.1:26657",
 		DefValue: "",
 	}
+	p2pKeyFlag = &pflag.Flag{
+		Name:     "p2p.key",
+		Usage:    "Custom p2p Ed25519 private key",
```

Comment:
Is there a reason to pass it directly in the command line rather than as a file?

Reply:
@jbowen93 Is there a reason for this from the docker side? IMO it should be fine to just pass a file path rather than the actual key itself. This common practice and if it's an ENV var issue, then just mount the priv key into the container and set the ENV var to the path.

Reply:
Yeah it's a bit sus to get a private key as a non-interactive parameter. If anything we shouldn't even give that option to prevent bad habits.

Reply:
Yes, this should be avoided, because command line history is usually stored on disk to .bash_history, etc. This means an unprivileged attacker could be able to extract the private key, even if the server operator only intended for the keys to be stored in RAM and not on disk.

Reply:
Good points! To resolve we need input from @jbowen93, he had some thoughts about this and ask me to implement this as a param to facilitate docker setup, IIRC.

One more thing is that this param is supposed to be hidden, meaning that it exists, but is not shown on `help`, so regular users wouldn't know about it unless they read the code or were told by someone.

Reply:
Apologies for delay on getting back to this. Feedback from Renay, John, and Mustafa is correct, we should be using files for this instead of env vars. That's my mistake requesting this from Hlib.

The current setup for [docker](https://github.com/celestiaorg/devops-test/blob/main/devnet/docker/bridge-docker-compose.yml#L21) and [k8s](https://github.com/celestiaorg/devops-test/blob/main/devnet/k8s/core-nodes/0/core0-deployment.yaml#L93) expects the key to exist at `{repo.path}/keys/OAZHALLLMV4Q`.

Reply:
https://github.com/celestiaorg/networks/issues/46 is the issue I'm using to track documenting best practices for external validators.